addons/mrp/models/mrp_production.py

```diff
@@ -402,6 +408,11 @@ def _compute_product_id(self):
             ):
                 production.product_id = bom.product_id or bom.product_tmpl_id.product_variant_id
 
+    @api.depends('product_id', 'product_id.uom_id', 'product_id.uom_ids', 'product_id.bom_ids', 'product_id.bom_ids.product_uom_id')
+    def _compute_allowed_uom_ids(self):
+        for production in self:
+            production.allowed_uom_ids = production.product_id.uom_id | production.product_id.uom_ids | production.product_id.bom_ids.product_uom_id
```

Comment:
Imho, that should be a method on `product` that gives you all allowed uoms for a single product, eventually overriden in modules that can add uoms to that pool.

Reply:
agreed, the spec specifics specific domains for specific model types, but I find it confusing + harder to maintain from the code point of view